web/client/epics/styleeditor.js

```diff
@@ -205,7 +205,25 @@ export const toggleStyleEditorEpic = (action$, store) =>
             if (!layer || layer && !layer.url) return Rx.Observable.empty();
 
             const geoserverName = findGeoServerName(layer);
-            if (!geoserverName) return Rx.Observable.empty();
+            if (!geoserverName) {
+                return getLayerCapabilities(layer)
```

Comment:
Should we avoid a getCapabilities request if the availableStyles property is listed inside the layer object?
Something like:

```js
...
if (layer.availableStyles) {
    return Rx.Observable.of(
        updateSettingsParams({ availableStyles: layer.availableStyles  }),
        loadedStyle()
    ); 
}
return getLayerCapabilities(layer)
...
```

Reply:
@AGMETEOR ^